Remove product display_as cell

Matt-Yorkley opened the pull request:
#### What? Why?

- Closes #11195

This bit of code was attempting to reference the master variant, which is now gone. Fixes an Angular scope/$eval error that shows up in the console on the admin products index page (and in Bugsnag).

#### What should we test?

No console errors on the admin products index page.

#### Release notes

<!-- Please select one for your PR and delete the other. -->

Changelog Category: Technical changes

<!-- Choose a pull request title above which explains your change to a
     a user of the Open Food Network app. -->

The title of the pull request will be included in the release notes.

#### Related
This is an alternative solution which
- closes https://github.com/openfoodfoundation/openfoodnetwork/pull/11196

mkllnk reviewed (approved):
Okay, the column itself doesn't need removing or replacing?

jibees reviewed (commented):
You removed the content of a column without removing the `<td/>`. Is there any particular reason?

dacook reviewed (commented):
This is an alternative solution which
- closes https://github.com/openfoodfoundation/openfoodnetwork/pull/11196

I can see that `display_as` only exists on variants, so it's correct that this column will be blank for product (to edit it, you need to expand and edit the variant):
![Screen Shot 2023-08-04 at 9 35 42 am](https://github.com/openfoodfoundation/openfoodnetwork/assets/4188088/f75d6bbc-7fa4-4635-9e28-4b0417143fbc)

As JB mentions on the other PR , we have more cleaning up to do of master variants. At the least for now, I think we can at least remove these related lines: https://github.com/openfoodfoundation/openfoodnetwork/blob/master/app/assets/javascripts/admin/bulk_product_update.js.coffee#L312-L320

I'll give it a go.

dacook commented:
@Matt-Yorkley I've added a new commit, would you be able to review this? If you're happy then I think we're all good for testing.

Matt-Yorkley commented:
> @Matt-Yorkley I've added a new commit, would you be able to review this?

Hmmm... it's introduced a number of test failures. I think the commit you've added is changing something very different to what this PR is aimed at achieving. I've dropped that commit for now.

rioug reviewed (approved):
:+1:

drummer83 commented:
Hi @Matt-Yorkley,
Thanks for this one. I have compared before and after staging the PR.

### Before staging
Several errors with `scope.$eval` were observed.
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/1790176/b7fd6ac2-19b9-4959-b5f2-06d41591549c)

This is the product list:
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/1790176/e06ae4e9-cfeb-46ed-a6f6-af1a072f1af8)

### After staging
There is only one error left. The `scope.$eval` errors are all gone. :heavy_check_mark: 
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/1790176/95e252ed-38f7-4a21-9b8a-306067b350f1)

The product list looks exactly the same: 
![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/1790176/71471881-640a-4b59-b32f-73cdb37a01f9)

### Conclusion
I don't see any issues with this one. Great! :rocket: 
Merging! :partying_face:
